File: 199-binary-tree-right-side-view/binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
void view(TreeNode* root ,vector<int>&rightview, int level )
{
    if(root==NULL)
    {
        return ;
    }
    if(rightview.size()== level)
    {
        rightview.push_back(root->val);
    }
    view( root->right ,rightview, level+1 );
    view(root->left ,rightview, level+1 );
}
    vector<int> rightSideView(TreeNode* root) {
        vector<int>rightview ;
        int level =0 ;
        view( root ,rightview, level );
        return rightview ;
    }
};

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution(object):
    def rightSideView(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: List[int]
        """
        res = []
        if root is None:
            return res
        q =deque([root])
        while q:
            curr_level = []
            curr_length = len(q)
            for _ in range(curr_length):
                node = q.popleft()
                curr_level = node.val

                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            res.append(curr_level) 
        return res


        